pkg/kubelet/config/config.go

```diff
@@ -235,6 +240,10 @@ func (s *podStorage) merge(source string, change interface{}) (adds, updates, de
 				ref.Annotations = make(map[string]string)
 			}
 			ref.Annotations[kubetypes.ConfigSourceAnnotationKey] = source
+			// ignore static pods
+			if ref.ResourceVersion != "" {
+				s.podStartupLatencyTracker.ObservedPodOnWatch(ref, time.Now())
```

Comment:
'when' value: can we set this to time.Now() stored when the object has been received from watch (as soon as possible) to reduce impact of possible process delay?

Or are you suggesting that this code is always happening directly after we receive the watch event?

Reply:
I see your point and I do agree, the time could (should?) be captured as soon as we observed the Pod Update, so most likely here: https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/config/apiserver.go#L64

On the other hand, the `kubetypes.PodUpdate` is a main/base struct (used in 54 places in 14 files), so adding an eventTimestamp is not such a small change - why don't we wait for some input from `sig-node` people?

I'm happy to make such a change, if there's a possibility, that could be a real time difference between those two options.